File: palindrome.py
# Define a function to check if the given string is palindrome or not
def is_palindrome(s):
    # Convert the string to lowercase and remove non-alphanumeric characters
    s = ''.join(char.lower() for char in s if char.isalnum())
    # Compare the string with its reverse
    return s == s[::-1]

def main():
    string = input("Enter a string to check if it's a palindrome: ")
    if is_palindrome(string):
        print("The string is a palindrome.")
    else:
        print("The string is not a palindrome.")

if __name__ == "__main__":
    main()

File: palindrome.rs
use std::io;

// Define a function to check if the given string is palindrome or not
fn is_palindrome(input: &str) -> bool {
    let input = input.trim().to_lowercase(); // Convert input to lowercase and remove leading/trailing whitespace
    let reversed = input.chars().rev().collect::<String>(); // Reverse the string
    input == reversed // Check if the original and reversed strings are equal
}

fn main() {
    println!("Enter a string to check if it's a palindrome:");
    
    let mut input = String::new();
    io::stdin().read_line(&mut input)
        .expect("Failed to read line");

    let input = input.trim(); // Remove newline character
    
    if is_palindrome(input) {
        println!("'{}' is a palindrome!", input);
    } else {
        println!("'{}' is not a palindrome.", input);
    }
}
